### PR TITLE
Various improvements to the DB code

### PR DESCRIPTION
**Pull request by Zabuzard**
Mostly shown by linter, some are:
* gitignore generated DB files (schema, ...)
* try-with-resources for SELECT query
* more generalized generics (? super ...)
* locks have to be opened before the try-block
* converted new Object() {...} into named helper class

### CODE REVIEW

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 1]
```diff
@@ -79,11 +80,10 @@ private void handleGetMessage(String message, MessageReceivedEvent event) {
...
         try {
-            Optional<StorageRecord> foundValue = database.read(ctx -> {
-                return ctx.selectFrom(Storage.STORAGE)
-                          .where(Storage.STORAGE.KEY.eq(key))
-                          .stream()
-                          .findFirst();
+            Optional<StorageRecord> foundValue = database.read(context -> {
+                try (var query = context.selectFrom(Storage.STORAGE)) {
+                    return query.where(Storage.STORAGE.KEY.eq(key)).stream().findFirst();
+                }
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:86]:
```suggestion
                return Optional.ofNullable(context.selectFrom(Storage.STORAGE)
                                                  .where(Storage.STORAGE.KEY.eq(key))
                                                  .fetchOne());
```
Maybe? That way we can avoid the annoying try there.

File: application/src/main/java/org/togetherjava/tjbot/db/Database.java [hunk 2]
```diff
@@ -109,13 +110,12 @@ public void write(CheckedConsumer<DSLContext, DataAccessException> action) {
      * @return whatever the handler returned#
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/db/Database.java:110]:
```suggestion
     * @return whatever the handler returned
```

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java (hunk 1 above)
Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:86]: The linter says that `context.selectFrom(Storage.STORAGE)` opens a resource that has to be closed. Does `fetchOne` avoid that?

Comment (I-Al-Istannen) [reply application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:86]: Yes https://blog.jooq.org/tag/autocloseable/

**Review by marko-radosavljevic — APPROVED** [review]

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**